File: SerpentineApi/SerpentineApi/Responses/GroupResponse.cs
namespace SerpentineApi.Responses;

public class GroupResponse : BaseResponse
{
    public string Name { get; set; } = null!;
    public int UnreadMessages { get; set; } = 0;
    public string ChannelName { get; set; } = null!;
    public Ulid ChannelId { get; set; }

    public bool RequiresOverage { get; set; } = false;

    public bool Public { get; set; } = true;
    public GroupAccessResponse? MyAccess { get; set; } 
    
    public MessageResponse? LastMessage { get; set; } 
}

File: serpentineclient/src/models/responses/group-response.ts
import {BaseResponse} from "@/models/responses/base-response.ts";
import {MessageResponse} from "@/models/responses/message-response.ts";
import {GroupAccessResponse} from "@/models/responses/group-access-response.ts";

export class GroupResponse extends BaseResponse {

    name: string = "";
    unreadMessages: number = 0;
    channelName: string = "";
    public : boolean = true;
    requiresOverage : boolean = false;
    channelId: string = "";
    myAccess: GroupAccessResponse | null = null;
    lastMessage: MessageResponse | null = null;



    constructor(init?: Partial<GroupResponse>) {
        super(init);
        Object.assign(this, init);
    }



}